coffeecard/CoffeeCard.Library/Utils/ProductUserGroupUtils.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+using CoffeeCard.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+
+public class ProductUserGroupUtils
+{
+    /// <summary>
+    /// Determines whether a product is a perk or not.
+    /// A product is a perk if it is not available for regular customers.
+    /// </summary>
+    /// <param name="context">The DbContext to use for querying the database.</param>
+    /// <param name="product">The product to check.</param>
+    /// <returns>True if the product is a perk, false otherwise.</returns>
+    public static bool isPerk(DbContext context, Product product)
```

Comment:
It does not make sense that a utility function should call the database. In the MapperService, the products have already been loaded from the Database, so it would be better to ensure that the User Groups are loaded when fetching the products.

The products are loaded in the `ProductService.GetProducts()` service. I suggest add the include for UserGroups here and then you can filter in your utility method.